cmd/virt-handler/virt-handler.go

```diff
@@ -83,10 +98,106 @@ type virtHandlerApp struct {
 	VirtShareDir            string
 	WatchdogTimeoutDuration time.Duration
 	MaxDevices              int
+
+	signingCertBytes []byte
+	serverCertBytes  []byte
+	serverKeyBytes   []byte
+	clientCertBytes  []byte
+	clientKeyBytes   []byte
+
+	virtCli   kubecli.KubevirtClient
+	namespace string
+
+	migrationTLSConfig *tls.Config
 }
 
 var _ service.Service = &virtHandlerApp{}
 
+func (app *virtHandlerApp) getSelfSignedCert() error {
+	var ok bool
+
+	generateCerts := false
+	secret, err := app.virtCli.CoreV1().Secrets(app.namespace).Get(virtHandlerCertSecretName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			generateCerts = true
+		} else {
+			return err
+		}
+	}
+
+	if generateCerts {
+		// Generate new certs if secret doesn't already exist
+		caKeyPair, _ := triple.NewCA("kubevirt.io")
+		serverKeyPair, _ := triple.NewServerKeyPair(
+			caKeyPair,
+			"kubevirt.io:system:node:virt-handler",
+			"",
+			"",
+			"",
+			nil,
+			nil,
+		)
+
+		clientKeyPair, _ := triple.NewClientKeyPair(caKeyPair,
+			"kubevirt.io:system:node:virt-handler",
+			nil,
+		)
+
+		app.serverKeyBytes = cert.EncodePrivateKeyPEM(serverKeyPair.Key)
+		app.serverCertBytes = cert.EncodeCertPEM(serverKeyPair.Cert)
+		app.clientKeyBytes = cert.EncodePrivateKeyPEM(clientKeyPair.Key)
+		app.clientCertBytes = cert.EncodeCertPEM(clientKeyPair.Cert)
+		app.signingCertBytes = cert.EncodeCertPEM(caKeyPair.Cert)
+
+		secret := k8sv1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      virtHandlerCertSecretName,
+				Namespace: app.namespace,
+				Labels: map[string]string{
+					v1.AppLabel: "virt-api-aggregator",
+				},
+			},
+			Type: "Opaque",
+			Data: map[string][]byte{
+				serverCertBytesValue:  app.serverCertBytes,
+				serverKeyBytesValue:   app.serverKeyBytes,
+				clientCertBytesValue:  app.clientCertBytes,
+				clientKeyBytesValue:   app.clientKeyBytes,
+				signingCertBytesValue: app.signingCertBytes,
```

Comment:
The only reason we'd need to save "signingCertBytes" is if we plan on generating new certs, which this pr doesn't do.

Reply:
We need it to verify that the peer certificate is signed by this ca.